src/ir/comp.rs

```diff
@@ -1078,8 +1078,13 @@ impl CompInfo {
             return None;
         }
 
+        if self.kind == CompKind::Union && self.fields().len() == 0 {
```

Comment:
nit: No need for `self.kind == CompKind::Union`, since we early-return for struts before.

Also, `self.fields().is_empty()`.

Reply:
fixed